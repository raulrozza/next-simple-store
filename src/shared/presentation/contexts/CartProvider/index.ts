export { default as useAddToCart } from './hooks/useAddToCart';
