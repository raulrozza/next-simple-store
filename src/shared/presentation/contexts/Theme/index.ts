export { default as useTheme } from './hooks/useTheme';
export { default as Theme } from './implementations/StyledComponentTheme';
