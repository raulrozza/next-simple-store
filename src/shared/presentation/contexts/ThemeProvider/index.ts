export { default as useTheme } from './hooks/useTheme';
export { default as ThemeProvider } from './implementations/StyledComponentTheme';
