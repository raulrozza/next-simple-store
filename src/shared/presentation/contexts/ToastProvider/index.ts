export { useToastProvider } from './context';
export { default as ToastProvider } from './implementations/ReactToastifyProvider';
