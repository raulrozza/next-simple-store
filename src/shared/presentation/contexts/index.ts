export * from './CartProvider';
export * from './ThemeProvider';
