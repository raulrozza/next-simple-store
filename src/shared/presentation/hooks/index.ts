import { trpc } from '@/shared/infra/http/trpc';

const { useQuery, useMutation } = trpc;

export { useQuery, useMutation };
