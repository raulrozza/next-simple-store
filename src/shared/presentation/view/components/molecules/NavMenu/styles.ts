import { transparentize } from 'polished';
import styled, { css } from 'styled-components';

interface ContainerProps {
    direction: 'row' | 'column';
}

const DIRECTION_STYLES = {
    row: css`
        flex-direction: row;
        justify-content: center;
        width: 100%;
        ${({ theme }) => css`
            gap: ${theme.layout.spacing(4)};
        `}
    `,
    column: css`
        flex-direction: column;
        height: 100%;
        ${({ theme }) => css`
            background-color: ${theme.palette.gray['0']};
        `}
    `,
};

export const Container = styled.ul<ContainerProps>`
    display: flex;
    list-style: none;

    ${({ direction }) => DIRECTION_STYLES[direction]}

    ${({ theme }) => css`
        padding: ${theme.layout.spacing(2)};
    `}
`;

export const ListItem = styled.li`
    background-color: transparent;
    border-radius: ${({ theme }) => theme.layout.borderRadius.sm};

    &:hover {
        background-color: ${({ theme }) =>
            transparentize(0.9, theme.palette.gray['100'])};
    }
`;
