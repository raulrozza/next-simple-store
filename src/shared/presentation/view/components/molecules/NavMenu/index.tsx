import { ComponentPropsWithoutRef, FC } from 'react';

import { NavLink } from '@/shared/presentation/view/components/atoms';

import { Container, ListItem } from './styles';

interface NavMenuItem
    extends Omit<
        ComponentPropsWithoutRef<typeof NavLink>,
        'children' | 'active'
    > {
    text: string;
}

interface NavMenuProps {
    items: NavMenuItem[];
    activeItem: string;
    direction?: 'column' | 'row';
    children?: never;
}

const NavMenu: FC<NavMenuProps> = ({
    items,
    activeItem,
    direction = 'column',
}) => (
    <Container direction={direction} role="menu">
        {items.map(item => (
            <ListItem key={item.href} role="menuitem">
                <NavLink
                    href={item.href}
                    icon={item.icon}
                    active={activeItem === item.href}
                >
                    {item.text}
                </NavLink>
            </ListItem>
        ))}
    </Container>
);

export default NavMenu;
