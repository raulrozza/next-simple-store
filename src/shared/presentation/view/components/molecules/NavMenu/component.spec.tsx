import { render, screen } from '@testing-library/react';

import { ThemeProvider } from '@/shared/presentation/contexts';

import NavMenu from './index';

describe('NavMenu', () => {
    it('should render the component correctly, displaying all the options', () => {
        render(
            <NavMenu
                items={[
                    {
                        href: '/',
                        icon: 'home',
                        text: 'Home',
                    },
                    {
                        href: '/products',
                        icon: 'list',
                        text: 'Products',
                    },
                    {
                        href: '/customers',
                        icon: 'people',
                        text: 'Customers',
                    },
                ]}
                activeItem="/customers"
            />,
            {
                wrapper: ThemeProvider,
            },
        );

        expect(screen.getByText('Home')).toBeInTheDocument();
        expect(screen.getByText('Products')).toBeInTheDocument();
        expect(screen.getByText('Customers')).toBeInTheDocument();
        expect(screen).toMatchSnapshot();
    });
});
