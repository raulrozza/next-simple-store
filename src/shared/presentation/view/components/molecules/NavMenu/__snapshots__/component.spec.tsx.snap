// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavMenu should render the component correctly, displaying all the options 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  gap: 0.25rem;
  padding: 0.25rem 0.125rem;
  color: #506C6D;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.5rem;
  font-family: 'Open Sans',sans-serif;
}

.c2:hover {
  color: #375758;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  gap: 0.25rem;
  padding: 0.25rem 0.125rem;
  color: #506C6D;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.5rem;
  font-family: 'Open Sans',sans-serif;
  color: #3D879D;
}

.c3:hover {
  color: #375758;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  background-color: #FFFFFF;
  padding: 0.5rem;
}

.c1 {
  background-color: transparent;
  border-radius: 0.25rem;
}

.c1:hover {
  background-color: rgba(230,234,234,0.1);
}

<body>
    <div>
      <ul
        class="c0"
        direction="column"
        role="menu"
      >
        <li
          class="c1"
          role="menuitem"
        >
          <a
            aria-selected="false"
            class="c2"
            href="/"
          >
            <svg
              aria-label="home icon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M946.5 505L534.6 93.4a31.93 31.93 0 0 0-45.2 0L77.5 505c-12 12-18.8 28.3-18.8 45.3 0 35.3 28.7 64 64 64h43.4V908c0 17.7 14.3 32 32 32H448V716h112v224h265.9c17.7 0 32-14.3 32-32V614.3h43.4c17 0 33.3-6.7 45.3-18.8 24.9-25 24.9-65.5-.1-90.5z"
              />
            </svg>
            Home
          </a>
        </li>
        <li
          class="c1"
          role="menuitem"
        >
          <a
            aria-selected="false"
            class="c2"
            href="/products"
          >
            <svg
              aria-label="list icon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M464 480H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h416c26.51 0 48 21.49 48 48v352c0 26.51-21.49 48-48 48zM128 120c-22.091 0-40 17.909-40 40s17.909 40 40 40 40-17.909 40-40-17.909-40-40-40zm0 96c-22.091 0-40 17.909-40 40s17.909 40 40 40 40-17.909 40-40-17.909-40-40-40zm0 96c-22.091 0-40 17.909-40 40s17.909 40 40 40 40-17.909 40-40-17.909-40-40-40zm288-136v-32c0-6.627-5.373-12-12-12H204c-6.627 0-12 5.373-12 12v32c0 6.627 5.373 12 12 12h200c6.627 0 12-5.373 12-12zm0 96v-32c0-6.627-5.373-12-12-12H204c-6.627 0-12 5.373-12 12v32c0 6.627 5.373 12 12 12h200c6.627 0 12-5.373 12-12zm0 96v-32c0-6.627-5.373-12-12-12H204c-6.627 0-12 5.373-12 12v32c0 6.627 5.373 12 12 12h200c6.627 0 12-5.373 12-12z"
              />
            </svg>
            Products
          </a>
        </li>
        <li
          class="c1"
          role="menuitem"
        >
          <a
            aria-selected="true"
            class="c3"
            href="/customers"
          >
            <svg
              aria-label="people icon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7 14s-1 0-1-1 1-4 5-4 5 3 5 4-1 1-1 1H7zm4-6a3 3 0 1 0 0-6 3 3 0 0 0 0 6z"
              />
              <path
                d="M5.216 14A2.238 2.238 0 0 1 5 13c0-1.355.68-2.75 1.936-3.72A6.325 6.325 0 0 0 5 9c-4 0-5 3-5 4s1 1 1 1h4.216z"
                fill-rule="evenodd"
              />
              <path
                d="M4.5 8a2.5 2.5 0 1 0 0-5 2.5 2.5 0 0 0 0 5z"
              />
            </svg>
            Customers
          </a>
        </li>
      </ul>
    </div>
  </body>,
  "container": .c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  gap: 0.25rem;
  padding: 0.25rem 0.125rem;
  color: #506C6D;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.5rem;
  font-family: 'Open Sans',sans-serif;
}

.c2:hover {
  color: #375758;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  gap: 0.25rem;
  padding: 0.25rem 0.125rem;
  color: #506C6D;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.5rem;
  font-family: 'Open Sans',sans-serif;
  color: #3D879D;
}

.c3:hover {
  color: #375758;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  background-color: #FFFFFF;
  padding: 0.5rem;
}

.c1 {
  background-color: transparent;
  border-radius: 0.25rem;
}

.c1:hover {
  background-color: rgba(230,234,234,0.1);
}

<div>
    <ul
      class="c0"
      direction="column"
      role="menu"
    >
      <li
        class="c1"
        role="menuitem"
      >
        <a
          aria-selected="false"
          class="c2"
          href="/"
        >
          <svg
            aria-label="home icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M946.5 505L534.6 93.4a31.93 31.93 0 0 0-45.2 0L77.5 505c-12 12-18.8 28.3-18.8 45.3 0 35.3 28.7 64 64 64h43.4V908c0 17.7 14.3 32 32 32H448V716h112v224h265.9c17.7 0 32-14.3 32-32V614.3h43.4c17 0 33.3-6.7 45.3-18.8 24.9-25 24.9-65.5-.1-90.5z"
            />
          </svg>
          Home
        </a>
      </li>
      <li
        class="c1"
        role="menuitem"
      >
        <a
          aria-selected="false"
          class="c2"
          href="/products"
        >
          <svg
            aria-label="list icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M464 480H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h416c26.51 0 48 21.49 48 48v352c0 26.51-21.49 48-48 48zM128 120c-22.091 0-40 17.909-40 40s17.909 40 40 40 40-17.909 40-40-17.909-40-40-40zm0 96c-22.091 0-40 17.909-40 40s17.909 40 40 40 40-17.909 40-40-17.909-40-40-40zm0 96c-22.091 0-40 17.909-40 40s17.909 40 40 40 40-17.909 40-40-17.909-40-40-40zm288-136v-32c0-6.627-5.373-12-12-12H204c-6.627 0-12 5.373-12 12v32c0 6.627 5.373 12 12 12h200c6.627 0 12-5.373 12-12zm0 96v-32c0-6.627-5.373-12-12-12H204c-6.627 0-12 5.373-12 12v32c0 6.627 5.373 12 12 12h200c6.627 0 12-5.373 12-12zm0 96v-32c0-6.627-5.373-12-12-12H204c-6.627 0-12 5.373-12 12v32c0 6.627 5.373 12 12 12h200c6.627 0 12-5.373 12-12z"
            />
          </svg>
          Products
        </a>
      </li>
      <li
        class="c1"
        role="menuitem"
      >
        <a
          aria-selected="true"
          class="c3"
          href="/customers"
        >
          <svg
            aria-label="people icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7 14s-1 0-1-1 1-4 5-4 5 3 5 4-1 1-1 1H7zm4-6a3 3 0 1 0 0-6 3 3 0 0 0 0 6z"
            />
            <path
              d="M5.216 14A2.238 2.238 0 0 1 5 13c0-1.355.68-2.75 1.936-3.72A6.325 6.325 0 0 0 5 9c-4 0-5 3-5 4s1 1 1 1h4.216z"
              fill-rule="evenodd"
            />
            <path
              d="M4.5 8a2.5 2.5 0 1 0 0-5 2.5 2.5 0 0 0 0 5z"
            />
          </svg>
          Customers
        </a>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
